zerver/lib/bugdown/fenced_code.py

```diff
@@ -360,6 +403,23 @@ def format_quote(self, text: str) -> str:
             quoted_paragraphs.append("\n".join("> " + line for line in lines if line != ''))
         return "\n\n".join(quoted_paragraphs)
 
+    def format_spoiler(self, header: str, text: str) -> str:
+        output = []
+        header_div_open_html = '<div class="zpoiler-block"><div class="zpoiler-header">'
+        toggle_button_html = '<a class="zpoiler-button" aria-expanded="false">' \
+            '<span class="zpoiler-arrow zpoiler-button-closed"></span></a>'
+        header_text = header
+        end_header_start_content_html = '</div><div class="zpoiler-content' \
+            ' zpoiler-content-closed" aria-hidden="true">'
+        footer_html = '</div></div>'
```

Comment:
We should do the HTML generation using the ElementTree API, like the rest of our backend markdown processor does. (`bugdown/__init__.py`).

Reply:
I think this won't work here (and currently is not used for the latex handling above either) because we need the output of that to partially be treated as already processed HTML (and not rendered as plain text by bugdown) and the part of it (the header and content text) to still be processed by bugdown.

This is why we're calling the stash function (here and in the JS version of this code) on the HTML but not on the header or content text. I may have missed a feature available in ElementTree, though!

Reply:
OK, makes sense.  We don't do so in the adjacent code for similar features, so I think you're right.